### PR TITLE
Second Restructure

### PR DESCRIPTION
**Pull request by kyle-stytch**
Linear Tickets:
- [MOB-20](https://linear.app/stytch/issue/MOB-20) - Android app link support (really this just provides a way of configuring Stytch by passing in a `Uri` object, rather than a `scheme` string and a `host` string).
- [MOB-32](https://linear.app/stytch/issue/MOB-32) - Run unit tests automatically via GitHub Actions
- [MOB-38](https://linear.app/stytch/issue/MOB-38) - Replace `loginOrInvite` endpoint with `loginOrCreate` + `create_user_as_pending` flag set to true
- [MOB-40](https://linear.app/stytch/issue/MOB-40) - Add test app in Android repo

This is a PR containing a few different things, to avoid having to keep adding branches on top of one another for small changes.

### CODE REVIEW

File: .github/workflows/runOnGithub.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# .github/workflows/runOnGitHub.yml
```
Comment (kyle-stytch) [inline .github/workflows/runOnGithub.yml:1]: GitHub Actions workflow file

File: sdk/src/main/java/com/stytch/sdk/Stytch.kt [hunk 2]
```diff
@@ -172,11 +128,6 @@ public class StytchEvent private constructor(
     }
 }
 
-public enum class StytchLoginMethod {
-    LoginOrSignUp,
-    LoginOrInvite,
-}
-
```
Comment (kyle-stytch) [inline sdk/src/main/java/com/stytch/sdk/Stytch.kt:179]: Replaced with loginOrSignUp always, but with the option to set the `create_user_as_pending` flag

File: settings.gradle [hunk 3]
```diff
@@ -1,2 +1,3 @@
 include ':sdk'
-rootProject.name = "Sample"
\ No newline at end of file
+rootProject.name = "Sample"
+include ':testapp'
```
Comment (kyle-stytch) [inline settings.gradle:3]: include new test app in root project

File: testapp/src/main/java/com/stytch/testapp/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            showTitle = false
+            showSubtitle = false
+        }
+
+        loginButton = findViewById(R.id.launch_stytch_ui_button)
+
+        loginButton.setOnClickListener {
+            val launchStytchIntent = Intent(this, StytchMainActivity::class.java)
+            startActivity(launchStytchIntent)
+        }
```
Comment (kyle-stytch) [inline testapp/src/main/java/com/stytch/testapp/MainActivity.kt:49]: Logic here taken from my sample app, I'll expand it to be more thorough for testing later.

**Review by ethan-stytch — APPROVED** [review]

**Pull request merged**